src/main/kotlin/com/ajaxproject/telegrambot/bot/BankInfoBot.kt

```diff
@@ -1,36 +1,49 @@
 package com.ajaxproject.telegrambot.bot
 
-import com.ajaxproject.telegrambot.bot.commands.AbstractCommand
+import com.ajaxproject.telegrambot.bot.model.UserRequest
+import com.ajaxproject.telegrambot.bot.service.UserSessionService
 import org.slf4j.LoggerFactory
-import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Component
 import org.telegram.telegrambots.bots.TelegramLongPollingBot
 import org.telegram.telegrambots.meta.api.objects.Update
-
 @Component
 class BankInfoBot(
-    @Qualifier("commandsMap")
-    private val commandsMapping: Map<String, AbstractCommand>,
+    val dispatcher: Dispatcher,
+    val userSessionService: UserSessionService,
 ) : TelegramLongPollingBot() {
-
-    @Value("\${credentials.botUsername}")
+    @Value("\${bot.username}")
     private val botUsername: String = ""
 
-    @Value("\${credentials.botToken}")
+    @Value("\${bot.token}")
     private val token: String = ""
+    override fun getBotToken(): String = token
 
     override fun getBotUsername(): String = botUsername
-    override fun getBotToken(): String = token
 
     override fun onUpdateReceived(update: Update) {
-        if (update.hasMessage()) {
-            val text = update.message.text
-            commandsMapping[text]?.handle(update, this)
-        } else if (update.hasCallbackQuery()) {
-            val callBack = update.callbackQuery.data
-            log.debug("New request with callbackQuery: {}", callBack)
-            commandsMapping[callBack]?.handle(update, this)
+        if (update.hasMessage() && update.message.hasText()) {
+            val textFromUser = update.message.text
+
+            val userId = update.message.from.id
+            val userFirstName = update.message.from.firstName
+
+            log.info("[{}, {}] : {}", userId, userFirstName, textFromUser)
+
+            val chatId = update.message.chatId
+            val session = userSessionService.getSession(chatId)
+
+            val userRequest = UserRequest(
+                update = update,
+                userSession = session,
+                chatId = chatId
+            )
+            val dispatcher = dispatcher.dispatch(userRequest)
```

Comment:
what the type of `dispatcher` variable?

It's wierd to see that `.dispatch` function returns "dispatcher"(name of the variable)

Reply:
Ok, I've already seen the `.dispatch` function. 

"dispatch" name for variable with type `Boolean`  is really bad name